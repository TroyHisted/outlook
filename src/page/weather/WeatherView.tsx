import type { Forecast } from './types/Forecast';

// https://weather-gov.github.io/api/general-faqs
export default (props: { weatherDescription: string; forecast: Forecast }) => (
	<>
		<p>Current Weather: {props.weatherDescription}</p>
		{props.forecast && (
			<div>
				{props.forecast.properties.periods.map((period) => (
					<div key={period.startTime}>
						{period.startTime}
						{period.endTime}
						{period.detailedForecast}
					</div>
				))}
			</div>
		)}
	</>
);
