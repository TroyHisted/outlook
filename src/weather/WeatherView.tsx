import React from 'react';
import type { Forecast } from './types/Forecast';

// https://weather-gov.github.io/api/general-faqs
export default (props: { weatherDescription: string; forecast: Forecast }) => (
	<>
		<p>Current Weather: {props.weatherDescription}</p>
		{props.forecast && (
			<img
				src={props.forecast.properties.periods[0].icon}
				title={props.forecast.properties.periods[0].shortForecast}
			/>
		)}
	</>
);
