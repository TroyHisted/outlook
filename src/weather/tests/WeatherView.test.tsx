import renderer from 'react-test-renderer';
import WeatherView from '../WeatherView';

test('renders correctly with no forecast', () => {
	const tree = renderer
		.create(<WeatherView forecast={undefined} weatherDescription="" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

test('renders correctly with a forecast', () => {
	const tree = renderer
		.create(
			<WeatherView
				forecast={{
					properties: {
						updated: '2020-01-01T00:00:00+0',
						periods: [
							{
								number: 0,
								name: 'Tonight',
								startTime: '2020-01-01T00:00:00+0',
								endTime: '2020-01-01T00:00:00+0',
								isDaytime: true,
								temperature: 72,
								temperatureUnit: 'F',
								TemperatureTrend: 'steady',
								windSpeed: '10 mph',
								windDirection: 'e',
								icon: 'clouds.png',
								shortForecast: 'Partly Cloudy',
								detailedForecast:
									'Partly Cloudy with a chance of meatballs.',
							},
						],
					},
				}}
				weatherDescription="Partly Cloudy"
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
