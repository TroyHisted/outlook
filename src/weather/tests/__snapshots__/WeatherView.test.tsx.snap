// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a forecast 1`] = `
[
  <p>
    Current Weather: 
    Partly Cloudy
  </p>,
  <img
    src="clouds.png"
    title="Partly Cloudy"
  />,
]
`;

exports[`renders correctly with no forecast 1`] = `
<p>
  Current Weather: 
</p>
`;
